### PR TITLE
TST: fix tests

### PR DESCRIPTION
**Pull request by wasade**
* order of items was out of sync
* nosetests requires the test directory to have an __init__

### CODE REVIEW

**Comment by wasade** [comment]
@ElDeveloper, do you have any idea on why the tests are failing or why coverage is dumping so much extra crap? I can't tell if Travis is insane, or if I am:

```bash
(python27)19:39:55 mustached-octo-ironman@master$ git checkout fix_tests
Switched to branch 'fix_tests'
(python27)19:39:58 mustached-octo-ironman@fix_tests$ git status
# On branch fix_tests
nothing to commit, working directory clean
(python27)19:40:00 mustached-octo-ironman@fix_tests$ git diff origin/fix_tests | wc -l
       0
(python27)19:40:05 mustached-octo-ironman@fix_tests$ nosetests
....................
----------------------------------------------------------------------
Ran 20 tests in 4.372s

OK
```

**Comment by josenavas** [comment]
@wasade from a quick look it looks like you're missing a .coveragerc file to tell coveralls which files needs to include in the report. You can take a look at the one in qiita.

**Comment by wasade** [comment]
Bam, thanks! :)

**Comment by wasade** [comment]
...I've got a few other PRs to get in after this that are staged. Should make things a little better

**Comment by josenavas** [comment]
@wasade are the failures expected?

**Comment by wasade** [comment]
No, I cannot recreate locally. Had pinged @ElDeveloper but he wasn't sure
either

File: moi/test/test_group.py [hunk 1]
```diff
@@ -99,15 +99,15 @@ def __call__(self, data):
         self.obj.forwarder = fwd
 
         self.obj.action('add', ['d', 'e'])
-        self.assertEqual(fwd.result, [
+        self.assertEqual(sorted(fwd.result), sorted([
```
Comment (antgonza) [inline moi/test/test_group.py]: Should all of these be replaced by assertItemsEqual?

File: moi/test/test_job.py [hunk 2]
```diff
@@ -48,19 +54,25 @@ def foo(a, b, **kwargs):
...
 
         r_client.set(self.test_job_info['id'], json.dumps(self.test_job_info))
-        _redis_wrap(self.test_job_info, foo, 1, 2)
+        obs_ret = _redis_wrap(self.test_job_info, foo, 1, 2)
 
-        sleep(2)
+        sleep(1)
         obs = json.loads(r_client.get(self.test_job_info['id']))
         self.assertEqual(obs['result'], 3)
+        self.assertEqual(obs_ret, obs['result'])
```
Comment (antgonza) [inline moi/test/test_job.py]: I think line 61 is not necessary based on this line, right?

Comment (wasade) [inline moi/test/test_job.py]: `obs` is the result that is stored in redis, while `obs_ret` is the direct result of executing the method through `_redis_wrap`. This ensures that `_redis_wrap` is actually storing the expected result in redis.

File: moi/test/test_group.py (hunk 1 above)
Comment (wasade) [inline moi/test/test_group.py]: Good catch

File: moi/test/test_group.py [hunk 3]
```diff
@@ -99,15 +99,15 @@ def __call__(self, data):
         self.obj.forwarder = fwd
 
         self.obj.action('add', ['d', 'e'])
-        self.assertEqual(fwd.result, [
+        self.assertItemsEqual(fwd.result, [
             {'add': {u'id': u'd', u'name': u'other job', u'type': u'job'}},
             {'add': {u'id': u'e', u'name': u'other job e', u'type': u'job'}}])
         self.obj.action('remove', ['e', 'd'])
-        self.assertEqual(fwd.result, [
+        self.assertEqual(sorted(fwd.result), sorted([
```
Comment (antgonza) [inline moi/test/test_group.py]: Could you also change this?

File: moi/test/test_group.py [hunk 4]
```diff
@@ -144,9 +144,9 @@ def __call__(self, data):
 
     def test_action_add(self):
         resp = self.obj._action_add(['d', 'f', 'e'])
-        self.assertEqual(resp, [
+        self.assertEqual(sorted(resp), sorted([
```
Comment (antgonza) [inline moi/test/test_group.py]: And this?

Comment (wasade) [inline moi/test/test_group.py]: Snagged all of them

On Tue, May 19, 2015 at 10:33 AM, Antonio Gonzalez <notifications@github.com

File: moi/test/test_job.py (hunk 2 above)
Comment (antgonza) [inline moi/test/test_job.py]: OK, thanks.

**Comment by josenavas** [comment]
@wasade just curiosity, which was the source of the errors?

**Comment by wasade** [comment]
ipython cluster wasn't given enough time to actually start. Need to add in better state checking code

**Comment by josenavas** [comment]
thanks!

**Pull request merged**